### PR TITLE
feat: json output

### LINKED ISSUE #17: feature: Add json output

### PR DESCRIPTION
**Pull request by zkoppert**
# Pull Request

## Proposed Changes

fixes #17 

This pull request introduces two new files, `json_writer.py` and `test_json_writer.py`. The changes implement a new functionality to write contributor data to a JSON file.

### Author/Contributor

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing

### Reviewer

- [x] Label as either `bug`, `documentation`, `enhancement`, `infrastructure`, `maintenance` or `breaking`

### CODE REVIEW

**Comment by zkoppert** [comment]
@jmeridth Are we missing any data that you think is important to put into the json output?

**Comment by zkoppert** [comment]
example outputs for comparison:
[contributors.json](https://github.com/user-attachments/files/15314184/contributors.json)
[contributors.md](https://github.com/user-attachments/files/15314185/contributors.md)

File: contributors.py [hunk 1]
```diff
@@ -72,7 +73,16 @@ def main():
...
-    # write_to_json(contributors)
+    json_writer.write_to_json(
+        filename="contributors.json",
+        start_date=start_date,
+        end_date=end_date,
+        organization=organization,
+        repository_list=repository_list,
+        sponsor_info=sponsor_info,
+        link_to_profile=link_to_profile,
+        contributors=contributors,
```
Comment (jmeridth) [inline contributors.py:84]: I think contributors is the first argument, right?  first argument to method but last on output.  Do we want to make same?

**Review by jmeridth — COMMENTED** [review]
One question.  Output, I think we're good

File: contributors.py (hunk 1 above)
Comment (jmeridth) [reply contributors.py:84]: ah, doesn't matter.  You're calling out the argument names.  NM

**Review by jmeridth — APPROVED** [review]

File: contributors.py (hunk 1 above)
Comment (zkoppert) [reply contributors.py:84]: Yeah I like to do this for readability and so I don't have to remember the order when there are so many args. Hopefully that's not too confusing.

**Pull request merged**